src/main.rs

```diff
@@ -129,9 +129,15 @@ pub async fn main() {
 		log::info!("IPFS backed application client: {}\t{:?}", peer_id, addrs);
 	}
 
-	let block_header = rpc::get_chain_header(&cfg.full_node_rpc)
-		.await
-		.expect("failed to get latest block header of chain");
+	let block_header = match rpc::get_chain_header(&cfg.full_node_rpc)
+		.await {
+			Ok(head) => head, 
+			Err(e) => {
+				log::info!("❗❗failed to get latest block header of chain, check if connected to node \n{:?}",e);
+				return
```

Comment:
This `return` and the previous `expect` have similar effects at `main` function.
If you want to log any error instead of just panic, my suggestion is to create a inner function and log any error in one single place. Something like:

```Rust
pub async fn main() {
    do_main().await().map_err(|err_msg| log::info!("{}", err_msg));
}

pub async fn do_main() -> Result<(), String> {
  /// ... previous main where `unwrap` was replaced by `?` & `map_err(...)`.
}
```

Having said that, I am **not big fan of using `String` as error type** (in `Result<(), String>`), and **we should use** a better error handling like [thiserror crate](https://crates.io/crates/thiserror) and/or [Anyhow](https://github.com/dtolnay/anyhow).

Reply:
+1 for anyhow!